src/Featurizers/NormalizeFeaturizer.h

```diff
@@ -0,0 +1,373 @@
+// |
+// |  NormalizeTransformer
+// |
+// ----------------------------------------------------------------------
+template <typename IteratorRangeT>
+NormalizeTransformer<IteratorRangeT>::NormalizeTransformer(std::vector<std::double_t> norms) :
+    _norms(std::move(norms)),
+    _row(0) {
+        if (_norms.empty()) {
+            throw std::invalid_argument("Norms input to NormalizeTransformer is empty!");
+        }
+        for (auto const &norm : _norms) {
+            if (norm < 0) {
+                throw std::invalid_argument("Norms shouldn't be less than 0!");
+            }
+        }
+}
+
+template <typename IteratorRangeT>
+NormalizeTransformer<IteratorRangeT>::NormalizeTransformer(Archive &ar) :
+    NormalizeTransformer(
+        [&ar](void) {
+            // Version
+            std::uint16_t                   majorVersion(Traits<std::uint16_t>::deserialize(ar));
+            std::uint16_t                   minorVersion(Traits<std::uint16_t>::deserialize(ar));
+
+            if(majorVersion != 1 || minorVersion != 0)
+                throw std::runtime_error("Unsupported archive version");
+
+            // Data
+            std::vector<std::double_t> norms(Traits<std::vector<std::double_t>>::deserialize(ar));
+
+            return NormalizeTransformer<IteratorRangeT>(std::move(norms));
+        }()
+    ) {
+}
+
+template <typename IteratorRangeT>
+void NormalizeTransformer<IteratorRangeT>::save(Archive &ar) const /*override*/ {
+    // Version
+    Traits<std::uint16_t>::serialize(ar, 1); // Major
+    Traits<std::uint16_t>::serialize(ar, 0); // Minor
+
+    // Data
+    Traits<std::vector<std::double_t>>::serialize(ar, _norms);
+}
+
+template <typename IteratorRangeT>
+bool NormalizeTransformer<IteratorRangeT>::operator==(NormalizeTransformer const &other) const {
+
+#if (defined __clang__)
+#   pragma clang diagnostic push
+#   pragma clang diagnostic ignored "-Wfloat-equal"
+#endif
+
+    return this->_norms == other._norms;
+
+#if (defined __clang__)
+#   pragma clang diagnostic pop
+#endif
+
+}
+
+
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+template <typename IteratorRangeT>
+void NormalizeTransformer<IteratorRangeT>::execute_impl(IteratorRangeT const &input, typename BaseType::CallbackFunction const &callback) /*override*/ {
+    // unpack input to two iterators
+    IteratorType begin = std::get<0>(input);
+    IteratorType const & end = std::get<1>(input);
+
+    execute_impl(begin, end, callback, std::integral_constant<bool, Microsoft::Featurizer::Traits<ValueType>::IsNullableType>());
+}
+
+template <typename IteratorRangeT>
+void NormalizeTransformer<IteratorRangeT>::execute_impl(IteratorType &begin, IteratorType const &end, typename BaseType::CallbackFunction const &callback, std::true_type) {

```

Comment:
>NormalizeTransformer [](start = 5, length = 20)

The code in these 2 methods is almost identical, with the only difference being how the iterator is accessed (is it wrapped in GetNullableValue or not). Can you think of a way to implement this logic in a single function that can be called by both methods?